pyteal/ast/base64decode.py

```diff
@@ -0,0 +1,85 @@
+from typing import TYPE_CHECKING
+from enum import Enum
+
+from pyteal.types import TealType, require_type
+from pyteal.errors import verifyFieldVersion
+from pyteal.ir import TealOp, Op, TealBlock
+from pyteal.ast.expr import Expr
+from pyteal.ast.leafexpr import LeafExpr
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class Base64Encoding(Enum):
+    # fmt: off
+    #     id  |   name   | min version
+    url = (0, "URLEncoding", 7)
+    std = (1, "StdEncoding", 7)
+    # fmt: on
+
+    def __init__(self, id: int, name: str, min_version: int) -> None:
+        self.id = id
+        self.arg_name = name
+        self.min_version = min_version
+
+
+Base64Encoding.__module__ = "pyteal"
+
+
+class Base64Decode(LeafExpr):
```

Comment:
nit: this shouldn't be a `LeafExpr`, that class is meant for expressions which return a value and take no arguments, i.e. the "leaves" of the AST

Reply:
@jasonpaulos I attempted to address in https://github.com/algorand/pyteal/pull/452.  Can you have a look and resolve if the PR captures it?